frame/bounties/src/migrations/v4.rs

```diff
@@ -0,0 +1,275 @@
+		<T as frame_system::Config>::BlockWeights::get().max_block
+	} else {
+		log::warn!(
+			target: "runtime::bounties",
+			"Attempted to apply migration to v4 but failed because storage version is {:?}",
+			on_chain_storage_version,
+		);
+		0
+	}
+}
+
+/// Some checks prior to migration. This can be linked to
+/// [`frame_support::traits::OnRuntimeUpgrade::pre_upgrade`] for further testing.
+///
+/// Panics if anything goes wrong.
+pub fn pre_migration<T: pallet_bounties::Config, P: GetStorageVersion + 'static, N: AsRef<str>>(
+	old_pallet_name: N,
+	new_pallet_name: N,
+) {
+	let old_pallet_name = old_pallet_name.as_ref();
+	let new_pallet_name = new_pallet_name.as_ref();
+	let storage_prefix_bounties_count = pallet_bounties::BountyCount::<T>::storage_prefix();
+	let storage_prefix_bounties = pallet_bounties::Bounties::<T>::storage_prefix();
+	let storage_prefix_bounties_description =
+		pallet_bounties::BountyDescriptions::<T>::storage_prefix();
+	let storage_prefix_bounties_approvals = pallet_bounties::BountyApprovals::<T>::storage_prefix();
+	log_migration("pre-migration", storage_prefix_bounties_count, old_pallet_name, new_pallet_name);
+	log_migration("pre-migration", storage_prefix_bounties, old_pallet_name, new_pallet_name);
+	log_migration(
+		"pre-migration",
+		storage_prefix_bounties_description,
+		old_pallet_name,
+		new_pallet_name,
+	);
+	log_migration(
+		"pre-migration",
+		storage_prefix_bounties_approvals,
+		old_pallet_name,
+		new_pallet_name,
+	);
+
+	let old_pallet_prefix = twox_128(old_pallet_name.as_bytes());
+	let old_bounties_count_key =
+		[&old_pallet_prefix, &twox_128(storage_prefix_bounties_count)[..]].concat();
+
+	let old_bounties_key = [&old_pallet_prefix, &twox_128(storage_prefix_bounties)[..]].concat();
+	let old_bounties_description_key =
+		[&old_pallet_prefix, &twox_128(storage_prefix_bounties_description)[..]].concat();
+	let old_bounties_approvals_key =
+		[&old_pallet_prefix, &twox_128(storage_prefix_bounties_approvals)[..]].concat();
+	/*
+	assert!(storage::next_key(&old_bounties_count_key)
+		.map_or(true, |next_key| next_key.starts_with(&old_bounties_key) ||
+			next_key.starts_with(&old_bounties_description_key) ||
+			next_key.starts_with(&old_bounties_approvals_key)));
+	assert!(storage::next_key(&old_bounties_key)
+		.map_or(true, |next_key| next_key.starts_with(&old_bounties_count_key) ||
+			next_key.starts_with(&old_bounties_description_key) ||
+			next_key.starts_with(&old_bounties_approvals_key)));
+	assert!(storage::next_key(&old_bounties_description_key)
+		.map_or(true, |next_key| next_key.starts_with(&old_bounties_count_key) ||
+			next_key.starts_with(&old_bounties_key) ||
+			next_key.starts_with(&old_bounties_approvals_key)));
+	assert!(storage::next_key(&old_bounties_approvals_key)
+		.map_or(true, |next_key| next_key.starts_with(&old_bounties_key) ||
+			next_key.starts_with(&old_bounties_description_key) ||
+			next_key.starts_with(&old_bounties_count_key)));
+			*/
```

Comment:
I think we don't need to be so precise, ensuring that the new prefix only contains the storage_version is enough IMO